langchain-core/src/language_models/base.ts

```diff
@@ -474,46 +488,34 @@ export abstract class BaseLanguageModel<
   }
 
   withStructuredOutput?<
-    RunInput = BaseLanguageModelInput,
-    // prettier-ignore
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    RunOutput extends z.ZodObject<any, any, any, any> = z.ZodObject<any, any, any, any>
+    RunOutput extends z.infer<AnyObjectType> = z.infer<AnyObjectType>
   >({
     schema,
     name,
     method,
     includeRaw,
-  }: {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    schema: z.ZodEffects<RunOutput> | Record<string, any>;
-    name: string;
-    method?: "functionCalling" | "jsonMode";
-    includeRaw: true;
-  }): Runnable<RunInput, { raw: BaseMessage; parsed: RunOutput }>;
+  }: StructuredOutputMethodParams<RunOutput, false>): Runnable<
+    BaseLanguageModelInput,
+    RunOutput
+  >;
 
   withStructuredOutput?<
-    RunInput = BaseLanguageModelInput,
-    // prettier-ignore
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    RunOutput extends z.ZodObject<any, any, any, any> = z.ZodObject<any, any, any, any>
+    RunOutput extends z.infer<AnyObjectType> = z.infer<AnyObjectType>
   >({
     schema,
     name,
     method,
     includeRaw,
-  }: {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    schema: z.ZodEffects<RunOutput> | Record<string, any>;
-    name: string;
-    method?: "functionCalling" | "jsonMode";
-    includeRaw?: false;
-  }): Runnable<RunInput, RunOutput>;
+  }: StructuredOutputMethodParams<RunOutput, true>): Runnable<
+    BaseLanguageModelInput,
+    { raw: BaseMessage; parsed: RunOutput }
+  >;
 
   /**
    * Model wrapper that returns outputs formatted to match the given schema.
    *
    * @template {BaseLanguageModelInput} RunInput The input type for the Runnable, expected to be the same input for the LLM.
-   * @template {z.ZodObject<any, any, any, any>} RunOutput The output type for the Runnable, expected to be a Zod schema object for structured output validation.
+   * @template {AnyObjectType} RunOutput The output type for the Runnable, expected to be a Zod schema object for structured output validation.
```

Comment:
Can this be an object too now? Not just a zod object?

Also `@template` above should be dropped and one for `IncludeRaw` should be added